src/riak_test_escript.erl

```diff
@@ -373,6 +373,15 @@ print_summary(TestResults, CoverResult, Verbose) ->
     end,
     ok.
 
+proper_exit_status(TestResults) ->
+    FailCount = length(lists:filter(fun(X) -> proplists:get_value(status, X) =:= fail end, TestResults)),
```

Comment:
A list comprehension would be clearer here, and you also don't need to compute the length in the following case expression if you just match the empty list.